packages/client/src/generation/TSClient/common.ts

```diff
@@ -416,6 +416,16 @@ type _Record<K extends keyof any, T> = {
   [P in K]: T;
 };
 
+// cause typescript not to expand types and preserve names
+type NoExpand<T> = T extends unknown ? T : never;
```

Comment:
For my own understanding, when `T extends unknown` would not be true?

Reply:
It's always true

Reply:
It's completely artificial. TypeScript has a way (I cannot recall how they call this) to preserve type names and not expand them when wrapped inside more complex generic types. So here, i am improving DX by utilizing that behavior at our advantage. See with and without:

Without:
![Screenshot from 2022-10-17 06-00-57](https://user-images.githubusercontent.com/18401805/196226537-8288e48e-e1a6-4165-9c0f-b6de785ed76a.png)

With:
![Screenshot from 2022-10-17 06-00-38](https://user-images.githubusercontent.com/18401805/196226547-01d95704-a853-4a9b-9db6-a1ad33d916f1.png)

Reply:
Thanks for the explanations!

Reply:
Auto completion stays the same within the object.